libs/core/src/main/java/org/opensearch/core/compress/CompressorRegistry.java

```diff
@@ -78,6 +78,19 @@ public static Compressor compressor(final BytesReference bytes) {
         return null;
     }
 
+    /**
+     * @param bytes The bytes to check the compression for
+     * @return The detected compressor. If no compressor detected then return NoneCompressor.
+     */
+    public static Compressor getCompressor(final BytesReference bytes) {
```

Comment:
compressorForWritable otherwise it gets confusing with compressor method

Reply:
updated